### ISSUE TITLE
Resources Exhausted while Training on GPU

### ISSUE THREAD

**Issue by isaacdchan** [thtrieu/darkflow#466]
Trying to train on my own dataset, hitting Resources Exhausted and OOM even while running on something stupid like --gpu 0.01
But when training on cpu, each epoch takes around 2-3 minutes, which is infeasible.

Details:
6 classes
25 images (each image has about 10 labels)
Batch size: 4

I remember someone saying that training and overfitting on a very small dataset could be beneficial.

**Comment by DreadPiratePsyopus** [comment]
Which kind of GPU are you using? I have found this model uses a lot of RAM from GPU, but your batch size is already small.....so probably you have some driver issue or you need better hardware.

**Comment by Bharathgoku** [comment]
I am having the same problem, if i have reduced the batch size=8 and subdivisions=1.I am getting the following error
InternalError (see above for traceback): Blas SGEMM launch failed : m=1352, n=35, k=1024
[[Node: 22-convolutional_2 = Conv2D[T=DT_FLOAT, data_format="NHWC", padding="VALID", strides=[1, 1, 1, 1], use_cudnn_on_gpu=true, _device="/job:localhost/replica:0/task:0/device:GPU:0"](Pad_8, 22-convolutional/kernel/read)]]
	 [[Node: mul_17/_59 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device="/job:localhost/replica:0/task:0/device:GPU:0", send_device_incarnation=1, tensor_name="edge_1099_mul_17", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:CPU:0"]()]]
Please anyone help me with this issue.
I want to know whether the problem is with memory or with the driver. 
Details:
classes =2,
55 images,
epoch=100.
I am using Geforce 840M(2GB) graphic card and ubuntu 17.10 .
If memory of the graphic card is the problem then how much more memory is needed to train our own dataset.

**Comment by linydf** [comment]
@Bharathgoku  Excuse me, would you please tell me how did you solve this problem?

**Comment by DreadPiratePsyopus** [comment]
Hello,
I used a Nvidia Gtx1080 and was able to only use a batch size of 12. So probably you need much more GPU RAM. You can try using your cpu just to test if the code or your command is ok by adding: 
 --gpu 0

**Comment by SrikanthVelpuri** [comment]
Gtx1080 is not enough for running yolo with atleast 32 images per
batch....then it cant reach majority of people....darknet is lot more
efficient than this
.

On Mon, Apr 9, 2018 at 2:08 PM, DreadPiratePsyopus <notifications@github.com

**Comment by shubh2719** [comment]
how big is your Neural Net architecture? It depends a lot on it...And also how big is each image in your dataset?..... There might be a possibility that your ram is getting exhausted while loading the dataset

**Comment by Leo223** [comment]
Anyway, the final test would be, assigning to **1** to the argument **batch_size** in your model.fit, which means that the GPU will charge images in memory from 1 to one, if it still get the same error, you will have two options:
- Change your  Neuronal Network Architecture
- Run it in another GPU wihich has greater RAM

**Comment by Leo223** [comment]
Anyway, the final test would be, assigning to **1** to the argument **batch_size** in your model.fit, which means that the GPU will charge images in memory from 1 to one, if it still get the same error, you will have two options:
- Change your  Neuronal Network Architecture
- Run it in another GPU wihich has greater RAM
